docs/docs-beta/docs/guides/deployment/docker.md

```diff
@@ -1,3 +1,197 @@
+  <summary>Prerequisites</summary>
+- Familiarity with Docker
+</details>
+
+### Understand the Docker containers that run as part of a Dagster Docker deployment
+
+A typical Dagster Docker deployment includes the following containers:
+- A container for the Dagster web server
+- A container for the [Dagster daemon](/deployment/dagster-daemon)
+- A container for each [code location](/concepts/code-locations), which runs the code server for that code location
+- A container for each run
+
+The containers for the web server and daemon typically share the same image. Each code location typically has its own image, and that image is also used for runs launched for that code location.
+
+### Define a Docker image for the Dagster web server and daemon
+
+The Dagster web server and daemon are the two _host processes_ in a Dagster deployment. They typically each run in their own container, using the same Docker image. This image contains Dagster packages and configuration, but no user code.
+
+To build this Docker image, use a Dockerfile like the following, with a name like `Dockerfile_dagster`:
+
+```dockerfile
+FROM python:3.10-slim
+
+RUN pip install \
+    dagster \
+    dagster-graphql \
+    dagster-webserver \
+    dagster-postgres \
+    dagster-docker
+
+# Set $DAGSTER_HOME and copy dagster.yaml and workspace.yaml there
+ENV DAGSTER_HOME=/opt/dagster/dagster_home/
+
+RUN mkdir -p $DAGSTER_HOME
+
+COPY dagster.yaml workspace.yaml $DAGSTER_HOME
+
+WORKDIR $DAGSTER_HOME
+```
+
+`dagster-postgres` is included to enable using Postgres as your Dagster database. `dagster-docker` is included to enable using the Docker run launcher, which launches each run within a Docker container.
+
+You'll need to have a [workspace file (`workspace.yaml`)](/concepts/code-locations/workspace-files) file in the same directory as the Dockerfile, to tell the web server and daemon where to look for the code servers.
+
+You'll also need to have a [dagster.yaml](https://docs.dagster.io/deployment/dagster-instance) file in the same directory as the Dockerfile, to configure the instance.
+
+### Define a Docker image for each code location
+
+The Docker image for a [code location](/concepts/code-locations) is used for the container for the code server, and for the container for each run launched for that code location.
+
+To build a Docker image for a code location, use a Dockerfile like the following, with a name like `Dockerfile_code_location_1`:
+
+```dockerfile
+FROM python:3.10-slim
+
+RUN pip install \
+    dagster \
+    dagster-postgres \
+    dagster-docker
+
+# Add code location code
+WORKDIR /opt/dagster/app
+COPY directory/with/your/code/ /opt/dagster/app
+
+# Run dagster code server on port 4000
+
+EXPOSE 4000
+
+# CMD allows this to be overridden from run launchers or executors to execute runs and steps
+CMD ["dagster", "code-server", "start", "-h", "0.0.0.0", "-p", "4000", "-f", "definitions.py"]
+```
+
+### Write a docker-compose file for all the services
```

Comment:
`all the services` strikes me as vague - what exactly does this mean?

I don't think the heading needs to be overly granular, but more specific would be good.